src/mlpack/methods/kmeans/kill_empty_clusters.hpp

```diff
@@ -51,13 +51,18 @@ class KillEmptyClusters
       const size_t emptyCluster,
       const arma::mat& /* oldCentroids */,
       arma::mat& newCentroids,
-      arma::Col<size_t>& /* clusterCounts */,
+      arma::Col<size_t>& counts /* clusterCounts */,
       MetricType& /* metric */,
       const size_t /* iteration */)
   {
-    // Kill the empty cluster.
-    newCentroids.col(emptyCluster).fill(DBL_MAX);
-    return 0; // No points were changed.
+    // Remove the empty cluster.
+      if (emptyCluster < newCentroids.n_cols)
+      {
+          newCentroids.shed_col(emptyCluster);
+          counts.shed_row(emptyCluster);
+          return 0; // one cluster removed
```

Comment:
Should this be 1? Or can we remove the return value here and just return at the end, since we always return 0.